src/components/Card/Card-Vertical/Card-Vertical.jsx

```diff
@@ -10,8 +15,15 @@ return (sortPriceHighLow.length === 0) ? <div className="container-card">Oh noo!
     <div className="card card-vertical">
         <div className="img-content">
             <p className="text-badge">{products.textBadge}</p>
-            <button className="btn-wishlist">
-                <i className="fas fa-heart"></i>
+            <button className="btn-wishlist"  onClick={() => wishListHandler(products, wishListState, wishDispatch)}>
+            {wishListState.inWishlist = wishListState.wishItems.some((item) => {
+                                if (item.id === products.id) {
+                                    return true;
+                                }
+                                return false
+                            })
+                            }
+                            <i className={wishListState.inWishlist ? "fas fa-heart" : "far fa-heart"}></i>
```

Comment:
we can do the check of inwishlist outside the return area like above in the same file or in a seperate file which have some other utility functions too .anway it's also totally fine.